### PR TITLE
feat(cli): procedure front matter SSOT, context-build hints, and gate record ergonomics

### LINKED ISSUE #116: feat(cli): surface adapter procedure hints in context build
## Context

Today `rgd context build` emits `observation`, `state`, `routes`, `guards`, `knowledge`, and `diagnostics` per `docs/cli.md` and `pkg/schema/operational-context.json`. Adapters still route to `.reinguard/procedure/*.md` using human-oriented sources (historically ADR-0013 §4). That does not scale to **spec-driven** `rgd`: runtime truth must come from `.reinguard/` declarative config, not repo-specific ADR prose.

This issue adds **machine-readable, advisory** procedure hints to `context build` output so Adapters can open the right procedure file from JSON alone. Hints are derived from the **procedure front matter SSOT** (`applies_to.state_ids`) defined in **#117**; they do not replace procedures, approval gates, or resume artifacts (ADR-0015).

## Refs: ADR

ADR-0001, ADR-0008, ADR-0011, ADR-0013, ADR-0015

## ADR Impact

amend `docs/adr/0013-fsm-workflow-states-and-adapter-mapping.md` only as coordinated with **#117** (§4 table removal / design-record reframing); otherwise keep normative CLI detail in `docs/cli.md`

## Acceptance ↔ ADR

- Hints are derived from repository-declared procedure front matter, not invented or hard-coded in the CLI beyond reading that config.
- Output remains stateless and pull-based: hints reflect only the current observation, loaded config, and parsed procedures.
- Hints are advisory for Adapter consumption; they do not encode approval, resume, or chat rules.
- `docs/cli.md` and `pkg/schema/operational-context.json` stay aligned (ADR-0008).

## Definition of Done

- **JSON shape** (when `state.kind` is `resolved`): extend the `state` object (same shape family as `resolutionResult`) with an optional inline object:

  `procedure_hint`: `{ "procedure_id": "<id>", "path": "<repo-relative .md path>", "derived_from": "state_id" }`

  When `state.kind` is not `resolved`, **omit** `procedure_hint` (absent field, not `null`). Do **not** add route-level procedure hints in this issue (state-only; avoids ambiguity vs §2 route hints).

- Implement hint computation in the `context build` pipeline (`internal/rgdcli/rgdcli.go` and helpers) using `internal/procedure/` from **#117**.
- Update `pkg/schema/operational-context.json` (`resolutionResult` / `state` documentation) and `docs/cli.md` (`rgd context build`) with field semantics and stability notes (advisory; derived from front matter).
- Update Adapter / Semantics cross-links per **#117** DoD (operational routing via `state.procedure_hint`, not ADR §4 table).
- Add or extend tests under `internal/rgdcli/` (and `internal/procedure/` as needed) for representative resolved vs non-resolved outcomes.
- Run `rgd config validate` if any Semantics source is touched.

## Test plan

- Intent: for a fixed observation fixture and minimal procedure set, `context build` JSON includes the expected `state.procedure_hint` when `state.kind` is `resolved`, and omits it for non-resolved outcomes.
- Boundaries: drift between front matter and FSM `state_id` sets should fail `config validate` or tests loudly, not silently emit wrong hints.
- Go checks: `go test ./... -race`, `go vet ./...`, `golangci-lint run`, `rgd config validate`

## Linked issues

- **Combined implementation PR with #117** (SSOT + consumer).
- Part of **#59** (Phase 2 Epic — Track D).
- Extends **#21** (operational context + `context build` baseline).
- Evolves from **#112** (Adapter pointers; this issue adds machine-readable routing).
- Builds on **#72** (FSM + mapping v1).
- Related: **#118** (context build flags / exit codes / help — coordinate if tests touch the same code paths).
- Related: **#74** (golden fixtures may need `procedure_hint` expectations later).

## Notes

- Do not add orchestration commands that execute procedures; stay within operational-context enrichment.
- Avoid field names that imply substrate workflow position such as `next_command`.

### LINKED ISSUE #117: refactor(control): single machine-readable SSOT for ADR-0013 §4 adapter mapping
## Context

`rgd` is **spec-driven**: at runtime it reads declarative config under `.reinguard/`, not ADR prose or `docs/cli.md`. Other repositories that install `rgd` will have different ADRs (or none). Treating ADR-0013 section 4 as an operational lookup table therefore misaligns the product model: the durable `state_id` → primary procedure mapping must live in Semantics config.

Procedure files already declare `applies_to.state_ids` (and optionally `route_ids`) in YAML front matter under `.reinguard/procedure/`. Today `config.Load` does not parse those files; ADR-0013 §4 duplicates the mapping as Markdown. That split forces Adapters to join human prose with substrate JSON and increases drift risk.

**#112** aligned Adapter pointers toward ADR-0013 §4; this issue completes the shift by making **procedure front matter the single machine-readable SSOT** and reframing ADR-0013 §4 as a **design record** (why the mechanism exists), not a second source of truth.

## Refs: ADR

ADR-0001, ADR-0008, ADR-0011, ADR-0013

## ADR Impact

amend `docs/adr/0013-fsm-workflow-states-and-adapter-mapping.md` — delete the §4 state_id → procedure **table**; rewrite §4 to describe the mechanism (front matter + `rgd context build`); update §5 extension contract step for adapter mapping to reference procedure front matter updates instead of the table

## Acceptance ↔ ADR

- One authoritative mapping lives in `.reinguard/procedure/*.md` front matter (`applies_to.state_ids`); validated by `rgd config validate` (or equivalent) together with control rules.
- ADR-0013 remains the human explanation of FSM state semantics and design rationale; it does not duplicate the mapping as a normative table.
- No new workflow-brain or orchestration behavior is added to `rgd`.

## Definition of Done

- Add a dedicated Go package `internal/procedure/` that loads and parses procedure front matter from `.reinguard/procedure/*.md` (pattern aligned with `internal/knowledge/frontmatter.go`; do not fold into `config.Load` core structs).
- Extend `rgd config validate` to run procedure-mapping checks: e.g. duplicate `state_id` across procedures, `state_id` in front matter with no matching state rule, malformed front matter.
- Amend ADR-0013 §4 as above; update §5 “Extension contract” so “Adapter mapping” means updating affected procedure `applies_to.state_ids`, not editing a Markdown table.
- **Purge operational “read ADR-0013 §4 table for routing” instructions** from Semantics and Adapter docs in favor of `state.procedure_hint` from `rgd context build` and/or procedure paths under `.reinguard/procedure/`. Concrete touchpoints (implementation pass):
  - `.cursor/commands/rgd-next.md` (Route / Propose trace)
  - `.cursor/rules/reinguard-bridge.mdc` (Workflow)
  - `.reinguard/procedure/next-orchestration.md` (Reads, full-path proposal, loop Route step)
  - `.reinguard/knowledge/workflow--state-gate-guard-extension.md` (checklist item for mapping updates)
- Add tests for invalid or missing mapping, duplicate `state_id`, and representative valid loads.
- Run `rgd config validate` after Semantics or schema changes.

## Test plan

- Intent: validation fails on duplicate `state_id` mappings across procedures, malformed front matter, and orphan `state_id` values not backed by control state rules.
- Boundaries: repos with no `.reinguard/procedure/` directory degrade gracefully (no mapping errors unless required by policy); migration from “ADR table as narrative” to “front matter only” does not break `config validate` on a clean tree.
- Go checks: `go test ./... -race`, `go vet ./...`, `golangci-lint run`, `rgd config validate`

## Linked issues

- Companion / same PR as **#116** (procedure hints consume this SSOT).
- Part of **#59** (Phase 2 Epic — Track D: spec-driven adapter mapping).
- Evolves from **#112** (Adapter SSOT alignment; this issue moves SSOT from ADR prose to config).
- Builds on **#72** (FSM YAML + Adapter mapping v1 introduced the §4 table being retired).
- Related: **#74** (golden / E2E fixtures may need updates when validate or config surface changes).

## Notes

- SSOT stays under `.reinguard/` (Semantics), not under `.cursor/`.
- Do not introduce a parallel mapping file unless front matter proves insufficient; the chosen SSOT is procedure front matter.

### PR DESCRIPTION
**Pull request by k-shibuki**
## Summary

- Load and validate `.reinguard/procedure/*.md` YAML front matter as the machine-readable state-to-procedure mapping SSOT (`applies_to.state_ids` / `applies_to.route_ids`), and reject duplicate or unmapped state references during `rgd config validate`
- Surface `state.procedure_hint` in `rgd context build` when a resolved state matches procedure front matter, and tighten the operational-context schema and tests around when that hint is allowed
- Add repeatable inline `--check id:status:summary` support to `rgd gate record` alongside `--checks-file`, plus refresh the adapter resume artifact schema version handling

## Traceability

Closes #116
Closes #117

Refs: #116
Refs: #117

## Definition of Done

- [x] Procedure front matter is the machine-readable SSOT for adapter procedure mapping, with validation coverage for duplicates and orphaned state references
- [x] `rgd context build` emits `state.procedure_hint` only for resolved states that match procedure front matter, with schema and CLI tests updated
- [x] `rgd gate record` supports repeatable inline `--check` entries and merged check assertions are covered by tests
- [x] Resume artifact schema version handling is aligned with ADR-0015 and local gate proofs were refreshed on the latest head

## Test plan

1. `bash .reinguard/scripts/with-repo-local-state.sh -- go test ./...`
2. `bash .reinguard/scripts/with-repo-local-state.sh -- go vet ./...`
3. `bash .reinguard/scripts/with-repo-local-state.sh -- golangci-lint run`
4. `go run ./cmd/rgd config validate`
5. `bash .reinguard/scripts/with-repo-local-state.sh --home-subdir cr-home -- bash .reinguard/scripts/check-local-review.sh --base main --retry-on-rate-limit`

## Risk / Impact

- The changes touch procedure loading, operational-context schema validation, and gate-record CLI parsing, so a regression could affect adapter routing hints or runtime gate artifact authoring.
- Review fixes also tighten parser edge cases and schema expectations, which is low risk but broad enough to warrant full CLI and package verification.

## Rollback Plan

- Revert commit `4de808d` if the review-driven tightening causes regressions in procedure parsing, schema validation, or gate recording behavior.
- If needed, re-open the previous PR head while keeping the issue linkage and PR template sections intact.

## Exception

- Type:
- Justification:

### CODE REVIEW

File: internal/procedure/frontmatter.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	seen := make(map[string]string)
+	for _, v := range values {
+		key := v
+		if prev, ok := seen[key]; ok {
+			return fmt.Errorf("procedure: front matter: duplicate %s %q in procedure id %q (also %q)", kind, v, procedureID, prev)
+		}
+		seen[key] = v
+	}
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (k-shibuki) [reply internal/procedure/frontmatter.go:88]: Fixed in 4de808d. Simplified duplicate value reporting and switched the uniqueness check to a set-backed implementation.

File: internal/procedure/hint_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		t.Fatalf("got %+v", got)
+	}
+}
+
+func TestHintEntry_unknownState(t *testing.T) {
+	t.Parallel()
+	if HintEntry([]Entry{{ID: "x", StateIDs: []string{"A"}}}, "B", true, "r") != nil {
+		t.Fatal("expected nil")
+	}
+}
```
[the remark this answers is not in the document]
Comment (k-shibuki) [reply internal/procedure/hint_test.go]: Fixed in 4de808d. Consolidated the related HintEntry scenarios into one table-driven test with subtests for route match, mismatch, unresolved route, empty route scope, and unknown state.

File: internal/rgdcli/rgdcli_gate_test.go [hunk 3]
```diff
@@ -146,6 +146,97 @@ rules:
...
+		"--cwd", repo,
+		"--status", "pass",
+		"--producer-procedure", "implement",
+		"--check", "bad-format",
+		"local-verification",
+	})
+	if err == nil {
+		t.Fatal("expected error for malformed --check value")
+	}
+}
```
[the remark this answers is not in the document]
Comment (k-shibuki) [reply internal/rgdcli/rgdcli_gate_test.go:285]: Fixed in 4de808d. Expanded malformed --check coverage into subtests for missing separators, empty components, and other delimiter edge cases.

File: internal/rgdcli/rgdcli_context_test.go [hunk 4]
```diff
@@ -159,6 +159,90 @@ exit 1
...
+	}
+	var out map[string]any
+	if err := json.Unmarshal(buf.Bytes(), &out); err != nil {
+		t.Fatal(err)
+	}
+	state := mustMap(t, out["state"], "state")
+	if _, ok := state["procedure_hint"]; ok {
+		t.Fatalf("unexpected procedure_hint: %v", state["procedure_hint"])
+	}
+}
```
[the remark this answers is not in the document]
Comment (k-shibuki) [reply internal/rgdcli/rgdcli_context_test.go:209]: Fixed in 4de808d. The negative test now asserts that both the state and the primary route resolve as expected before verifying that procedure_hint is omitted.

File: internal/rgdcli/rgdcli_gate_test.go [hunk 5]
```diff
@@ -146,6 +146,97 @@ rules:
...
+
+	var out map[string]any
+	if err := json.Unmarshal(buf.Bytes(), &out); err != nil {
+		t.Fatalf("json: %v raw=%s", err, buf.String())
+	}
+	checks, ok := out["checks"].([]any)
+	if !ok || len(checks) != 2 {
+		t.Fatalf("expected 2 checks (1 from file + 1 inline), got %v", out)
+	}
+}
```
[the remark this answers is not in the document]
Comment (k-shibuki) [reply internal/rgdcli/rgdcli_gate_test.go:247]: Fixed in 4de808d. The merged-check assertion now validates both check IDs and their status and summary payloads instead of only checking the slice length.

File: pkg/schema/operational-context.json [hunk 6]
```diff
@@ -37,7 +37,18 @@
...
+          "type": "object",
+          "description": "Optional advisory mapping from resolved state_id to a procedure file (front matter SSOT). Omitted when state.kind is not resolved.",
+          "required": ["procedure_id", "path", "derived_from"],
+          "properties": {
+            "procedure_id": { "type": "string" },
+            "path": { "type": "string" },
+            "derived_from": { "type": "string", "const": "state_id" }
+          },
+          "additionalProperties": false
+        }
```
[the remark this answers is not in the document]
Comment (k-shibuki) [reply pkg/schema/operational-context.json:51]: Fixed in 4de808d. Added a schema guard that forbids procedure_hint whenever kind is not resolved, and locked that behavior with a focused compiler test.

**Comment by k-shibuki** [comment]
Disposition: By design.

Quoted finding:
> Docstring Coverage: warning that coverage is 16.98% against an 80% threshold.

Rationale:
This repository does not use a blanket docstring-percentage target as a merge gate. The changes in this PR are covered by focused tests and validation, and the review-driven follow-up kept documentation updates scoped to the user-facing and policy-facing surfaces that changed.

**Comment by k-shibuki** [comment]
@coderabbitai review

Follow-up applied in 4de808d: addressed the six inline findings, tightened the procedure_hint schema contract and parser edge cases, refreshed local verification and local CodeRabbit gates, and updated the PR template sections for policy compliance.

**Comment by k-shibuki** [comment]
@coderabbitai review

File: internal/rgdcli/rgdcli_gate_test.go [hunk 7]
```diff
@@ -146,6 +146,127 @@ rules:
...
+	}
+	checks, ok := out["checks"].([]any)
+	if !ok || len(checks) != 2 {
+		t.Fatalf("expected 2 checks, got %v", out)
+	}
+	first := checks[0].(map[string]any)
+	if first["id"] != "go-test" || first["status"] != "pass" || first["summary"] != "go test ./... -race" {
+		t.Fatalf("unexpected first check: %v", first)
+	}
+}
```
[the remark this answers is not in the document]
Comment (k-shibuki) [reply internal/rgdcli/rgdcli_gate_test.go:196]: Fixed in e2cfdf0. The success-path test now validates both inline checks with guarded map casts instead of asserting only the first entry.

File: internal/rgdcli/rgdcli_gate_test.go [hunk 8]
```diff
@@ -146,6 +146,127 @@ rules:
...
+				"--config-dir", cfgDir,
+				"--cwd", repo,
+				"--status", "pass",
+				"--producer-procedure", "implement",
+				"--check", v,
+				"local-verification",
+			})
+			if err == nil {
+				t.Fatalf("expected error for malformed --check value %q", v)
+			}
```
[the remark this answers is not in the document]
Comment (k-shibuki) [reply internal/rgdcli/rgdcli_gate_test.go:279]: Fixed in e2cfdf0. The malformed --check subtests now assert the specific parser contract message in addition to requiring a non-nil error.

**Comment by k-shibuki** [comment]
@coderabbitai review

Follow-up applied in e2cfdf0: tightened `internal/rgdcli/rgdcli_gate_test.go` so the inline `--check` success path validates both emitted checks via guarded object casts, and the malformed `--check` table test now asserts the specific parse-contract error text.

**Comment by k-shibuki** [comment]
@coderabbitai review
